### PR TITLE
layers: More err enum work in obj_tracker

### PR DESCRIPTION
**Pull request by karl-lunarg**
Change-Id: I299b68ec2e66d77812ce5dff0d457e328cdd6c5c

### CODE REVIEW

File: layers/object_tracker.cpp [hunk 1]
```diff
@@ -1578,31 +1583,30 @@ VKAPI_ATTR void VKAPI_CALL UpdateDescriptorSets(VkDevice device, uint32_t descri
...
                     for (uint32_t idx3 = 0; idx3 < pDescriptorWrites[idx1].descriptorCount; ++idx3) {
                         if (pDescriptorWrites[idx1].pImageInfo[idx3].imageView) {
                             skip_call |= ValidateObject(device, pDescriptorWrites[idx1].pImageInfo[idx3].imageView,
-                                                                       VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_VIEW_EXT, false);
+                                                        VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_VIEW_EXT, false, VALIDATION_ERROR_00943);
                         }
-                        if (pDescriptorWrites[idx1].pImageInfo[idx3].sampler) {
-                            skip_call |= ValidateObject(device, pDescriptorWrites[idx1].pImageInfo[idx3].sampler,
-                                                                       VK_DEBUG_REPORT_OBJECT_TYPE_SAMPLER_EXT, false);
+                        if (pDescriptorWrites[idx1].pImageInfo[idx3].imageLayout) {
```
Comment (mark-lunarg) [inline layers/object_tracker.cpp]: This looks funny.  VkImageLayouts aren't objects -- was this supposed to get changed from a VkSampler check?

File: layers/object_tracker.cpp [hunk 2]
```diff
@@ -675,8 +675,8 @@ VKAPI_ATTR VkResult VKAPI_CALL InvalidateMappedMemoryRanges(VkDevice device, uin
         if (pMemoryRanges) {
             for (uint32_t idx0 = 0; idx0 < memoryRangeCount; ++idx0) {
                 if (pMemoryRanges[idx0].memory) {
-                    skip_call |= ValidateObject(device, pMemoryRanges[idx0].memory,
-                                                               VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_MEMORY_EXT, false);
+                    skip_call |= ValidateObject(device, pMemoryRanges[idx0].memory, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_MEMORY_EXT,
```
Comment (tobine) [inline layers/object_tracker.cpp]: I believe this error is for a valid struct type, which will be checked by parameter_checker, should this be 00648 instead?

File: layers/object_tracker.cpp [hunk 3]
```diff
@@ -652,8 +652,8 @@ VKAPI_ATTR VkResult VKAPI_CALL FlushMappedMemoryRanges(VkDevice device, uint32_t
...
```
Comment (tobine) [inline layers/object_tracker.cpp]: I believe this error is for a valid struct type, which will be checked by parameter_checker, should this be 00648 instead?

File: layers/object_tracker.cpp (hunk 1 above)
Comment (tobine) [inline layers/object_tracker.cpp]: Yes, shouldn't parameter_checker check the layout?

File: layers/object_tracker.cpp (hunk 3 above)
Comment (karl-lunarg) [inline layers/object_tracker.cpp]: Yes.   Thanks, will fix.

File: layers/object_tracker.cpp (hunk 2 above)
Comment (karl-lunarg) [inline layers/object_tracker.cpp]: Yup

**Pull request merged**